### ISSUE TITLE
Räumliche Erschließung von übergangsweise parallel QIDs und Strings 

### ISSUE THREAD

**Issue by dr0i** [hbz/lobid-resources#998]
Result of hbz/nwbib#470:

New catalogued nwbib data will look like this:
>
         <datafield tag="700" ind1="n" ind2="1">
            <subfield code="a">https://nwbib.de/spatial#N05 Westfalen</subfield>
          </datafield>
          <datafield tag="700" ind1="n" ind2="1">
            <subfield code="a">https://nwbib.de/spatial#Q2742 Münster</subfield>
          </datafield>
          <datafield tag="700" ind1="n" ind2="1">
            <subfield code="a">https://nwbib.de/subjects#N844200 Denkmalpflege. Denkmalschutz</subfield>
          </datafield>

The `spatial` entries in the jsonld will be generated in two ways:
1. the old way: if an entry in `7001n1` doesn't start with "https:" lookup the literal in geo_nwbib and use the result to build the jsonld structure via ElasticsearchIndexer.java
2. the new way: if it starts with "https:" lookup the tsv map (consisting of all the entries in geo_nwbib) and build the data structure using the morph

@acka47 What to do with the `coverage` field, valued e.g. `"Köln | 99"`? See e.g. http://test.lobid.org/resources/HT015854197.json . This will be omitted going with 2)  I think (at leas the snippet above doesn't reflect these entries, and I assume nobody will catalogue them anymore, right?)), but be necessary when going with 1) which we must use in parallel for at least some time.

**Comment by acka47** [comment]
> What to do with the `coverage` field, valued e.g. `"Köln | 99"`? See e.g. http://test.lobid.org/resources/HT015854197.json . This will be omitted going with 2)  I think (at leas the snippet above doesn't reflect these entries, and I assume nobody will catalogue them anymore, right?)), but be necessary when going with 1) which we must use in parallel for at least some time.

Exctly, if entries start with `https`, the `coverage` field won't be necessary anymore. (That will also be a good way to check for already resources already updated in hbz01 which will then be those with `_exists_:spatial AND NOT _exists_:coverage`.

**Comment by acka47** [comment]
Thus, we have to build a comprehensive tsv file covering all Wikidata entities needed so that we get data (coordinates & type) for the `focus` object. There are three different source to gather the entries from:

1. The SPARQL query for the geo index ([/src/main/resources/getNwbibSubjectLocationsAsWikidataEntities.sparql](https://github.com/hbz/lobid-resources/blob/master/src/main/resources/getNwbibSubjectLocationsAsWikidataEntities.sparql)
2. The hand-written string2qid list at [/src/main/resources/string2wikidata.tsv](https://github.com/hbz/lobid-resources/blob/master/src/main/resources/string2wikidata.tsv) that can now easily be created and updated with the SPARQL query from https://github.com/hbz/nwbib/issues/468#issuecomment-493189871
3. 33 entries that already have been part of NWBib spatial classification where a focus statement has been added to the SKOS file, see https://github.com/hbz/lobid-vocabs/commit/2dbcd44af7968092a158d33c1392bac517e96802#diff-a7e9c11f735c098bc8b7bb474747fc93.

Next steps: add the 33 QIDs from 3.) to the SPARQL query for 2.), adjust the SPARQL query for 1.) to be concatenated with the rest.

**Comment by acka47** [comment]
So I added the 34 (!) IDs from 3.) to the SPARQL query with eeade06, however, this will not be enough for adding the focus object. The reason is that something like this will be in the data: `https://nwbib.de/spatial#n03` while the corresponding QID to derive the `focus` information from is Q152243. We will have to think about a solution for this.

**Comment by acka47** [comment]
I also adjusted [/src/main/resources/getNwbibSubjectLocationsAsWikidataEntities.sparql](https://github.com/hbz/lobid-resources/blob/master/src/main/resources/getNwbibSubjectLocationsAsWikidataEntities.sparql) with 5492362. Now the results of both SPARQL queries can easily be concatenated for one TSV file to be used in the transformation process.

**Comment by acka47** [comment]
> something like this will be in the data: `https://nwbib.de/spatial#n03` while the corresponding QID to derive the `focus` information from is Q152243. We will have to think about a solution for this.

I will add a map URI to QID so that @dr0i can look it up.

**Comment by acka47** [comment]
Here is the map. Note that notation `70` has two WD entities as `focus`.

```tsv
1	Q1198
3	Q152243
5	Q8614
10	Q462011
12	Q72931
13	Q2036208
14	Q4194 
16	Q580471
18	Q881875
20	Q151993
22	Q153464
24	Q445609
28	Q152356 
32	Q1380992
33	Q1381014
34	Q1413205
42	Q7904317
44	Q836937
45	Q641138
46	Q249428
47	Q152420
48	Q708742
57	Q698162
62	Q657241
63	Q649192
64	Q650645
65	Q697254
66	Q514557
68	Q700198
69	Q573290
70	Q14551680,Q835382
76	Q153943
77	Q829718
```

**Comment by acka47** [comment]
+1 Everything looks good. As discussed offline, we will have to think about the broader regions (e.g. Eifel, Weserbergland, or Nordrhein-Westfalen itself) that have one geo point attached. it does not make sense to use those geo coordinates for the "result map". So we have to think about a way to handle this (ignoring these coordinates or not storing them to begin with).

**Comment by dr0i** [comment]
Deployed to production, closed.

**Comment by acka47** [comment]
Reopening. From https://github.com/hbz/nwbib/issues/470#issuecomment-541045997:

> Die Art der Speicherung von URI plus String in 700n wird ja nun anders ablaufen als in .https://github.com/hbz/nwbib/issues/470#issuecomment-483588151 dargestellt. Die endgültige Fassung mit Ablage von String und URI in unterschiedlichen Unterfeldern ist im [Wiki](https://service-wiki.hbz-nrw.de/pages/viewpage.action?pageId=339607588#URIsstattGSWundNotationeninhbz01&Lokalsystemen-Nachher) dokumentiert, hier das Beispiel:

```xml
<datafield tag="700" ind1="n" ind2="1">
    <subfield code="a">Ruhrgebiet</subfield>
    <subfield code="0">https://nwbib.de/spatial#N20</subfield>
</datafield>
<datafield tag="700" ind1="n" ind2="1">
    <subfield code="a">Duisburg</subfield>
    <subfield code="0">https://nwbib.de/spatial#Q2100</subfield>
</datafield>
<datafield tag="700" ind1="n" ind2="1">
    <subfield code="a">Essen</subfield>
    <subfield code="0">https://nwbib.de/spatial#Q2066</subfield>
</datafield>
<datafield tag="700" ind1="n" ind2="1">
    <subfield code="a">Einzelne Autoren (Primärliteratur)</subfield>
    <subfield code="0">https://nwbib.de/subjects#N768010</subfield>
</datafield>
```

We will have to update the respective file in hbz01XmlClobs.tar.bz2 and adjust the morph accordingly.

**Comment by dr0i** [comment]
This should be resolved with #1036. Although there is no way of real testing, as there is only so small a designed test case, I am going to deploy it to not have any conflicts in the morph when other work is done there.
Should be reopened when real data is coming in and behaves in a bad manner.
